### PR TITLE
Sw 102

### PR DESCRIPTION
**Pull request by DaniekFjaes**
image_2023-11-27_125859353

Still W.I.P and has a few issues:
- Needs some specific formatting for lower resolutions, right now it looks subpar
- Missing the ability to change quantity on basket items
- Basket items are currently just a placeholder, until someone actually makes the basket site
- Needs to give a warning when incorrect information is entered.
- Price isn't calculated because it should (probably?) be fetched from a service

### CODE REVIEW

File: src/app/webshop/checkout/checkout.component.ts [hunk 1]
```diff
@@ -7,11 +7,64 @@ import { FormGroup, FormControl } from '@angular/forms';
 })
 export class CheckoutComponent {
   checkoutForm = new FormGroup({
```
Comment (Thekid2002) [inline src/app/webshop/checkout/checkout.component.ts]: Use formbuilder

File: src/app/webshop/checkout/checkout.component.ts [hunk 2]
```diff
@@ -7,11 +7,64 @@ import { FormGroup, FormControl } from '@angular/forms';
...
   });
 
+  deliveryMethods = [
+    { value: 'pickup', cost: 0, label: 'Afhentning i Butik' },
+    { value: 'postnord', cost: 59, label: 'Postnord Hjemmelevering' },
+    { value: 'gls', cost: 59, label: 'GLS' },
+    { value: 'dao', cost: 59, label: 'DAO' }
+  ];
+
+  addressFields = [
```
Comment (Thekid2002) [inline src/app/webshop/checkout/checkout.component.ts]: Use wizard service : OR remove 🗡️

File: src/app/webshop/checkout/checkout.component.ts [hunk 3]
```diff
@@ -7,11 +7,64 @@ import { FormGroup, FormControl } from '@angular/forms';
...
+  }
+
+  // Should probably be in a basket/item service
+  calculateTotalPrice(basketContent: { name: string, price: string }[]): number {
+    let totalPrice = 0;
+
+    for (const item of basketContent) {
+      const itemPrice = parseFloat(item.price);
+
+      if (!isNaN(itemPrice)) {
```
Comment (Thekid2002) [inline src/app/webshop/checkout/checkout.component.ts]: Hent pris fra API <3

File: src/app/webshop/checkout/checkout.component.html [hunk 4]
```diff
@@ -1,18 +1,79 @@
...
-    <mat-form-field appearance="fill">
-      <mat-label for="lastName">Last Name: </mat-label>
-      <input matInput id="lastName" type="text" formControlName="lastName">
-    </mat-form-field>
-    <mat-form-field appearance="fill">
-      <mat-label for="address">Your Address: </mat-label>
-      <input matInput id="address" type="text" formControlName="address">
-    </mat-form-field>
+<app-front-basic-layout>
+  <div fxLayout = "row" fxFlexFill>
```
Comment (Thekid2002) [inline src/app/webshop/checkout/checkout.component.html]: No space in equal

**Review by Thekid2002 — CHANGES_REQUESTED** [review]

File: src/app/webshop/checkout/checkout.component.ts (hunk 3 above)
Comment (DaniekFjaes) [reply src/app/webshop/checkout/checkout.component.ts]: Jeg får vel itemDTOs fra basket, så giver det mening at hente priserne igen, når jeg allerede (burde) have dem?

File: src/app/webshop/checkout/checkout.component.ts [hunk 5]
```diff
@@ -1,18 +1,50 @@
...
+    });
+  }
+  deliveryMethods = [
+    { value: 'pickup', cost: 0, label: 'Afhentning i Butik' },
+    { value: 'postnord', cost: 59, label: 'Postnord Hjemmelevering' },
+    { value: 'gls', cost: 59, label: 'GLS' },
+    { value: 'dao', cost: 59, label: 'DAO' }
+  ];
+  // Placeholder. This must be fetched from the actual basket, so a service should probably be made to get that information
+  basketContent = [
```
Comment (Thekid2002) [inline src/app/webshop/checkout/checkout.component.ts]: DEn skal hentes fra orderservice

File: src/app/webshop/checkout/checkout.component.ts [hunk 6]
```diff
@@ -1,18 +1,50 @@
...
+    { name: 'God Gammeldags Rødvin', price: '300', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'God Gammeldags Rødvin', price: '300', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'God Gammeldags Rødvin', price: '300', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'God Gammeldags Rødvin', price: '300', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'God Gammeldags Rødvin', price: '300', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'Viktor Special', price: '15', imageUrl: 'assets/PeanutNoar.jfif' }
+  ];
   createCheckout():void{
-    console.log(this.checkoutForm.value);
+    console.log(this.checkoutForm);
```
Comment (Thekid2002) [inline src/app/webshop/checkout/checkout.component.ts]: We do not console.log in production code (unless errors) <3

File: src/app/webshop/checkout/checkout.component.ts [hunk 7]
```diff
@@ -1,18 +1,50 @@
...
-    address: new FormControl(''),
-  });
+export class CheckoutComponent implements OnInit{
+  checkoutForm: FormGroup | undefined;
 
+  constructor(private fb: FormBuilder) { }
+  ngOnInit() : void {
+    this.checkoutForm = this.fb.group({
+      // Define your form controls here using FormBuilder
+      // Example: 'name': [initialValue, [validators]]
```
Comment (Thekid2002) [inline src/app/webshop/checkout/checkout.component.ts]: Remove comments

File: src/app/webshop/checkout/checkout.component.ts [hunk 8]
```diff
@@ -1,18 +1,50 @@
...
+  basketContent = [
+    { name: 'Ugandisk Vin', price: '200', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'God Gammeldags Rødvin', price: '300', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'God Gammeldags Rødvin', price: '300', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'God Gammeldags Rødvin', price: '300', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'God Gammeldags Rødvin', price: '300', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'God Gammeldags Rødvin', price: '300', imageUrl: 'assets/PeanutNoar.jfif' },
+    { name: 'Viktor Special', price: '15', imageUrl: 'assets/PeanutNoar.jfif' }
+  ];
   createCheckout():void{
```
Comment (Thekid2002) [inline src/app/webshop/checkout/checkout.component.ts]: Add spacing and comments + should be public

File: src/app/webshop/checkout/checkout.component.ts [hunk 9]
```diff
@@ -1,18 +1,50 @@
...
-    console.log(this.checkoutForm.value);
+    console.log(this.checkoutForm);
+  }
+  goToPayment():void{
```
Comment (Thekid2002) [inline src/app/webshop/checkout/checkout.component.ts]: Add spacing and comment + should be public

**Review by Thekid2002 — CHANGES_REQUESTED** [review]

**Review by Thekid2002 — APPROVED** [review]

**Pull request merged**